packages/@uppy/transloadit/types/index.d.ts

```diff
@@ -2,124 +2,154 @@ import type { PluginOptions, UppyFile, BasePlugin } from '@uppy/core'
+  assembly_url: string
+  assembly_ssl_url: string
+  uppyserver_url: string
+  companion_url: string
+  websocket_url: string
+  tus_url: string
+  bytes_received: number
+  bytes_expected: number
+  upload_duration: number
+  client_agent?: string
+  client_ip?: string
+  client_referer?: string
+  transloadit_client: string
+  start_date: string
+  upload_meta_data_extracted: boolean
+  warnings: any[]
+  is_infinite: boolean
+  has_dupe_jobs: boolean
+  execution_start: string
+  execution_duration: number
+  queue_duration: number
+  jobs_queue_duration: number
+  notify_start?: any
+  notify_url?: string
+  notify_status?: any
+  notify_response_code?: any
+  notify_duration?: any
+  last_job_completed?: string
+  fields: Record<string, any>
+  running_jobs: any[]
+  bytes_usage: number
+  executing_jobs: any[]
+  started_jobs: string[]
+  parent_assembly_status: any
+  params: string
+  template?: any
+  merged_params: string
+  uploads: FileInfo[]
+  results: Record<string, Result[]>
+  build_id: string
+  error?: string
+  stderr?: string
+  stdout?: string
+  reason?: string
+}
 
-  interface AssemblyParameters {
-    auth: {
-      key: string,
-      expires?: string
-    }
-    template_id?: string
-    steps?: { [step: string]: Record<string, unknown> }
-    notify_url?: string
-    fields?: { [name: string]: number | string }
+interface AssemblyParameters {
+  auth: {
+    key: string
+    expires?: string
   }
+  template_id?: string
+  steps?: { [step: string]: Record<string, unknown> }
+  notify_url?: string
+  fields?: { [name: string]: number | string }
+}
 
 interface AssemblyOptions {
-    params: AssemblyParameters
-    fields?: { [name: string]: number | string }
-    signature?: string
-  }
+  params?: AssemblyParameters
+  fields?: { [name: string]: number | string }
+  signature?: string
+}
 
-interface TransloaditOptionsBase extends PluginOptions {
-    service?: string
-    errorReporting?: boolean
-    waitForEncoding?: boolean
-    waitForMetadata?: boolean
-    importFromUploadURLs?: boolean
-    alwaysRunAssembly?: boolean
-    locale?: TransloaditLocale
-    limit?: number
+interface Options extends PluginOptions {
+  service?: string
+  errorReporting?: boolean
+  waitForEncoding?: boolean
+  waitForMetadata?: boolean
+  importFromUploadURLs?: boolean
+  alwaysRunAssembly?: boolean
+  locale?: TransloaditLocale
+  limit?: number
 }
 
-// Either have a getAssemblyOptions() that returns an AssemblyOptions, *or* have them embedded in the options
-export type TransloaditOptions = TransloaditOptionsBase &
-    (
-      | {
-          getAssemblyOptions?: (file: UppyFile) => AssemblyOptions | Promise<AssemblyOptions>
-        }
-      | AssemblyOptions)
+export type TransloaditOptions = Options &
+  (
+    | {
+        assemblyOptions?: AssemblyOptions | ((file: UppyFile) => Promise<AssemblyOptions>)
+        getAssemblyOptions?: never
+        /** @deprecated use `assemblyOptions` instead */
+        params?: never
```

Comment:
Why is `params` marked as deprecated and `getAssemblyOptions ` is not if they are both marked as `never`?

Reply:
No reason, I just forgot one comment